pykokkos/lib/ufuncs.py

```diff
@@ -959,17 +960,166 @@ def multiply(viewA, viewB):
+    tid: int = team_member.league_rank()
+    def row_fill(i: int):
+        viewOut[tid][i] = scalar
+    pk.parallel_for(pk.TeamThreadRange(team_member, cols), row_fill)
+    
+@pk.workunit
+def stretch_fill_impl_1d_into_2d(team_member, cols, viewIn, viewOut):
+    tid: int = team_member.league_rank()
+    def row_fill(i: int):
+        viewOut[tid][i] = viewIn[i]
+    pk.parallel_for(pk.TeamThreadRange(team_member, cols), row_fill)
+
+@pk.workunit
+def stretch_fill_impl_2d(team_member, inner_its, col_wise, viewIn, viewOut):
+    tid: int = team_member.league_rank()
+    def inner_fill(i: int):
+        if col_wise:
+            viewOut[i][tid] = viewIn[i][0]
+        else:
+            viewOut[tid][i] = viewIn[0][i]
+    pk.parallel_for(pk.TeamThreadRange(team_member, inner_its), inner_fill)
+
+        
+
+def broadcast_view(val, viewB):
+    """
+    Broadcasts val onto viewB, returns the "stretched" version of viewA
+
+    Parameters
+    ----------
+    val : pykokkos view or Scalar
+            View or scalar to be broadcasted (is shorter and compatible in dimensions).
+    viewB : pykokkos view
+            View to be broadcasted onto (is longer and compatible in dimensions).
+
+    Returns
+    -------
+    out : pykokkos view
+           Broadcasted version of viewA.
+
+    """
+    if len(viewB.shape) > 2:
+        raise NotImplementedError("Broadcasting is only supported upto 2D views")
+
+    is_view = False
+    if isinstance(val, ViewType):
+        for dim in val.shape:
+            if dim != 1:
+                is_view = True
+        
+        if not is_view:
+            val = val[0] if len(val.shape) == 1 else val[0][0]
+
+    if is_view:
+        if not check_broadcastable_impl(val, viewB) or not val.shape < viewB.shape:
+            raise ValueError("Incompatible broadcast")
+        if not val.dtype == viewB.dtype: 
+            raise ValueError("Broadcastable views must have same dtypes")
+
+    out = pk.View([dim for dim in viewB.shape], viewB.dtype)
+
+    if is_view:
+        # if both 2D
+        if len(val.shape) == 2: #viewB must be 2 because of the val.shape < viewB.shape check
+            # figure which orientation is val (row or col)
+            col_wise = 1 if val.shape[1] == 1 else 0
+            inner_its = viewB.shape[0] if col_wise else viewB.shape[1]
+            outer_its = viewB.shape[1] if col_wise else viewB.shape[0]
+            policy = pk.TeamPolicy(outer_its, pk.AUTO)
+            pk.parallel_for(policy, stretch_fill_impl_2d, inner_its=inner_its, col_wise=col_wise, viewIn=val, viewOut=out)
+        else: # 1d to 2D
+            policy = pk.TeamPolicy(out.shape[0], pk.AUTO)
+            pk.parallel_for(policy, stretch_fill_impl_1d_into_2d, cols=viewB.shape[1], viewIn=val, viewOut=out)
+            
+        return out
+
+    # scalar
+
+    if len(viewB.shape) == 1:
+        out_1d = pk.View([viewB.shape[1] if len(viewB.shape)==2 else viewB.shape[0]])
```

Comment:
Something is off with your logic here. The condition to enter this block is `len(viewB.shape) == 1`, but you are checking if it is equal to 2 when building the shape